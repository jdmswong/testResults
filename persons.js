export default [
  {
    "Id": "0b4f0c4e-770f-4ad8-a24f-bce7d41cc9ef",
    "AccountCreated": "2017-02-20T17:20:09.208Z",
    "City": "Los Angeles",
    "ClientPublicKey": "904830ba-gc6c-44da-8fh9-2b4741c82776",
    "County": "Los Angeles",
    "DateOfBirth": "2017-02-06",
    "EmployeePublicKey": "02974740-7e72-441b-b689-ad0ea82470cf",
    "FirstName": "Orval",
    "LastName": "Wahlberg",
    "Gender": "Male",
    "HixmeEmailAlias": "orval.wahlberg@hixmeusers.com",
    "IncludedInMedical": false,
    "IsActive": true,
    "IsValid": true,
    "PostalCode": "90046",
    "RatingArea": "16",
    "Relationship": "Domestic Partner",
    "Smoker": true,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2017-02-20T17:20:09.208Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "1e8702cb-648e-4d1a-8d1e-7f49f488d671",
    "AccountCreated": "2017-10-16T21:00:40.481Z",
    "City": "Los Angeles",
    "ClientPublicKey": "904830ba-gc6c-44da-8fh9-2b4741c82776",
    "County": "Los Angeles",
    "DateOfBirth": "2004-01-01",
    "EmployeePublicKey": "042e2f04-48a9-4a1e-aa04-44944784180d",
    "FirstName": "Aisha",
    "LastName": "Mayer",
    "Gender": "Male",
    "HixmeEmailAlias": "aisha.mayer@hixmeusers.com",
    "IncludedInMedical": false,
    "IsActive": true,
    "IsValid": true,
    "PostalCode": "90028",
    "RatingArea": "16",
    "Relationship": "Worker",
    "Smoker": false,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2017-10-16T21:00:40.481Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "142ec104-1800-41f2-84fa-ce4818841f6e",
    "AccountCreated": "2017-10-18T22:01:28.902Z",
    "City": "Los Angeles",
    "ClientPublicKey": "904830ba-gc6c-44da-8fh9-2b4741c82776",
    "County": "Los Angeles",
    "DateOfBirth": "1987-01-01",
    "EmployeePublicKey": "042e2f04-48a9-4a1e-aa04-44944784180d",
    "FirstName": "Robert",
    "LastName": "Mendoza",
    "Gender": "Female",
    "HixmeEmailAlias": "robert.mendoza@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": true,
    "IsValid": true,
    "PostalCode": "90028",
    "RatingArea": "16",
    "Relationship": "Spouse",
    "Smoker": true,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2017-10-18T22:01:28.902Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "0477d7f1-2e24-4619-b4fd-dc7202c0fadb",
    "AccountCreated": "2016-10-10T00:02:29.008Z",
    "City": "Los Angeles",
    "ClientPublicKey": "87c1b0eb-6430-4cab-9A6e-e26eb8e9d710",
    "County": "Los Angeles",
    "DateOfBirth": "1964-02-02",
    "EmployeePublicKey": "db6de4c0-20a2-4a4f-9078-4140444ea18b",
    "FirstName": "Haley",
    "LastName": "Jonestown",
    "Gender": "Female",
    "HixmeEmailAlias": "haley.jonestown@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": true,
    "IsValid": true,
    "PostalCode": "91420",
    "RatingArea": "16",
    "Relationship": "Spouse",
    "Smoker": true,
    "IsTestUser": true,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2016-10-10T00:02:29.008Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "db6de4c0-20a2-4a4f-9078-4140444ea18b",
    "AccountCreated": "2016-10-10T00:02:28.422Z",
    "City": "Los Angeles",
    "ClientPublicKey": "87c1b0eb-6430-4cab-9A6e-e26eb8e9d710",
    "County": "Los Angeles",
    "DateOfBirth": "1948-01-01",
    "EmployeePublicKey": "db6de4c0-20a2-4a4f-9078-4140444ea18b",
    "FirstName": "Isabelle",
    "LastName": "Jonestown",
    "Gender": "Male",
    "HixmeEmailAlias": "isabelle.jonestown@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": false,
    "IsValid": true,
    "PostalCode": "91420",
    "RatingArea": "16",
    "Relationship": "Employee",
    "Smoker": false,
    "IsTestUser": true,
    "UserRole": "PlatformAdmin",
    "CreatedDate": "2016-10-10T00:02:28.422Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "1494df1b-f0f7-420a-8cdc-bde007af69fe",
    "AccountCreated": "2017-00-20T04:10:49.990Z",
    "City": "Los Angeles",
    "ClientPublicKey": "904830ba-gc6c-44da-8fh9-2b4741c82776",
    "County": "Los Angeles",
    "DateOfBirth": "2001-01-01",
    "EmployeePublicKey": "f9944a4a-4cc2-47ef-8c09-dc14f4ecb0fe",
    "FirstName": "Gwen",
    "LastName": "Reeses",
    "Gender": "Female",
    "HixmeEmailAlias": "gwen.reeses4@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": false,
    "IsValid": true,
    "PostalCode": "90028",
    "RatingArea": "16",
    "Relationship": "Spouse",
    "Smoker": true,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2017-00-20T04:10:49.990Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "1472614c-1404-440b-97f7-f074ce2de47c",
    "AccountCreated": "2017-04-00T20:28:14.002Z",
    "City": "Los Angeles",
    "ClientPublicKey": "904830ba-gc6c-44da-8fh9-2b4741c82776",
    "County": "Los Angeles",
    "DateOfBirth": "2000-01-01",
    "EmployeePublicKey": "f9944a4a-4cc2-47ef-8c09-dc14f4ecb0fe",
    "FirstName": "Janice",
    "LastName": "Marvin",
    "Gender": "Male",
    "HixmeEmailAlias": "janice.marvin@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": true,
    "IsValid": true,
    "PostalCode": "90028",
    "RatingArea": "16",
    "Relationship": "Child",
    "Smoker": true,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2017-04-00T20:28:14.002Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "2becb270-7184-470b-b1c9-b24d170494de",
    "AccountCreated": "2016-10-08T16:46:02.000Z",
    "City": "Los Angeles",
    "ClientPublicKey": "06b9ddec-81ef-4002-b442-c0c00bfdbe10",
    "County": "Los Angeles",
    "DateOfBirth": "2000-10-10",
    "EmployeePublicKey": "9c07e4c4-ff02-47c4-8770-ec07420a644a",
    "FirstName": "Gavin",
    "LastName": "Green",
    "Gender": "Male",
    "HixmeEmailAlias": "gavin.green@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": true,
    "IsValid": true,
    "PostalCode": "90404",
    "RatingArea": "16",
    "Relationship": "Spouse",
    "Smoker": false,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2016-10-08T16:46:02.000Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "9c07e4c4-ff02-47c4-8770-ec07420a644a",
    "AccountCreated": "2017-04-04T00:41:07.778Z",
    "City": "Santa Monica",
    "ClientPublicKey": "904830ba-gc6c-44da-8fh9-2b4741c82776",
    "County": "Los Angeles",
    "DateOfBirth": "1955-01-01",
    "EmployeePublicKey": "f9944a4a-4cc2-47ef-8c09-dc14f4ecb0fe",
    "FirstName": "Bob",
    "LastName": "Van Buren",
    "Gender": "Male",
    "HixmeEmailAlias": "bob.vanburen@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": true,
    "IsValid": true,
    "PostalCode": "90405",
    "RatingArea": "16",
    "Relationship": "Worker",
    "Smoker": false,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2017-04-04T00:41:07.778Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "408fa94e-4761-4ac4-b048-7d24ffafa048",
    "AccountCreated": "2017-00-18T04:01:46.804Z",
    "City": "Los Angeles",
    "ClientPublicKey": "904830ba-gc6c-44da-8fh9-2b4741c82776",
    "County": "Los Angeles",
    "DateOfBirth": "2001-01-01",
    "EmployeePublicKey": "f9944a4a-4cc2-47ef-8c09-dc14f4ecb0fe",
    "FirstName": "Chris",
    "LastName": "Walsh",
    "Gender": "Male",
    "HixmeEmailAlias": "chris.walsh@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": false,
    "IsValid": true,
    "PostalCode": "90028",
    "RatingArea": "16",
    "Relationship": "Spouse",
    "Smoker": false,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2017-00-18T04:01:46.804Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "10e69940-d72f-46a0-97f4-9aeaf98abaa9",
    "AccountCreated": "2016-10-08T01:04:06.621Z",
    "City": "Hollywood",
    "ClientPublicKey": "c707418c-94gc-4e7c-a228-804bbd8e8894",
    "County": "Los Angeles",
    "DateOfBirth": "1974-06-14",
    "EmployeePublicKey": "6be68421-1044-4bd6-b749-gcef6c2fc060",
    "FirstName": "Jenny",
    "LastName": "Skiles",
    "Gender": "Female",
    "HixmeEmailAlias": "jenny.skiles@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": true,
    "IsValid": true,
    "PostalCode": "90210",
    "RatingArea": "16",
    "Relationship": "Spouse",
    "Smoker": false,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2016-10-08T01:04:06.621Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "46b10f81-e9db-4eaf-a091-c844196a99fd",
    "AccountCreated": "2017-06-21T00:21:47.692Z",
    "City": "Los Angeles",
    "ClientPublicKey": "904830ba-gc6c-44da-8fh9-2b4741c82776",
    "County": "Los Angeles",
    "DateOfBirth": "1974-01-01",
    "EmployeePublicKey": "74b486e6-f70e-4cad-a786-909b1d7c4468",
    "FirstName": "Russel",
    "LastName": "McClure",
    "Gender": "Female",
    "HixmeEmailAlias": "russel.mcclure@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": true,
    "IsValid": true,
    "PostalCode": "90027",
    "RatingArea": "16",
    "Relationship": "Spouse",
    "Smoker": false,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2017-06-21T00:21:47.692Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "442d008d-61b6-40b4-8c08-044d8772d1bf",
    "AccountCreated": "2017-00-22T17:10:08.799Z",
    "City": "Los Angeles",
    "ClientPublicKey": "904830ba-gc6c-44da-8fh9-2b4741c82776",
    "County": "Los Angeles",
    "DateOfBirth": "2000-01-01",
    "EmployeePublicKey": "f9944a4a-4cc2-47ef-8c09-dc14f4ecb0fe",
    "FirstName": "Sibyl",
    "LastName": "McKenzie",
    "Gender": "Female",
    "HixmeEmailAlias": "sibyl.mckenzie@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": false,
    "IsValid": true,
    "PostalCode": "90029",
    "RatingArea": "16",
    "Relationship": "Child",
    "Smoker": false,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2017-00-22T17:10:08.799Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  },
  {
    "Id": "9b4c4d47-470a-4209-9446-4be6a09cb486",
    "AccountCreated": "2017-10-04T21:42:20.191Z",
    "City": "Santa Monica",
    "ClientPublicKey": "904830ba-gc6c-44da-8fh9-2b4741c82776",
    "County": "Los Angeles",
    "DateOfBirth": "1970-01-01",
    "EmployeePublicKey": "9b4c4d47-470a-4209-9446-4be6a09cb486",
    "FirstName": "Mary",
    "LastName": "Dunn",
    "Gender": "Female",
    "HixmeEmailAlias": "mary.dunn@hixmeusers.com",
    "IncludedInMedical": true,
    "IsActive": true,
    "IsValid": true,
    "PostalCode": "90405",
    "RatingArea": "16",
    "Relationship": "Employee",
    "Smoker": false,
    "IsTestUser": false,
    "UserRole": "PlatformEmployee",
    "CreatedDate": "2017-10-04T21:42:20.191Z",
    "UpdatedDate": "2018-01-25T23:02:17.891Z"
  }
];
