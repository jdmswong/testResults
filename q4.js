/*
Generate a new array of worker objects that have all the same data they
  have now, but with modified email addresses. The new form of their email
  address should be:

 "first_letter_of_first_name.first_three_letters_of_last_name.the_users_date_of_birth_year@newhixmeusers.com"

There are three main components—with periods between them each—and a "new"
  domain name at the end:

first_letter_of_first_name
first_three_letters_of_last_name
the_users_date_of_birth_year
 */

export default persons => persons.map(person => ({
  ...person,
  HixmeEmailAlias: `${person.FirstName.charAt(0)}.${person.LastName.substring(0, 3)}.${person.DateOfBirth.substring(0, 4)}@newhixmeusers.com`,
}))