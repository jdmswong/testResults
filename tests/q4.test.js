import q4 from '../q4';
import persons from '../persons';

test('empty set', () => {
  const sampleData = [];
  expect(q4(sampleData)).toMatchSnapshot();
});

test('returns expected value', () => {
  expect(q4(persons).map(e => e.HixmeEmailAlias)).toMatchSnapshot();
});
