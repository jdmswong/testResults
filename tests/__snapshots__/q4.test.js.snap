// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty set 1`] = `Array []`;

exports[`returns expected value 1`] = `
Array [
  "O.Wah.2017@newhixmeusers.com",
  "A.May.2004@newhixmeusers.com",
  "R.Men.1987@newhixmeusers.com",
  "H.Jon.1964@newhixmeusers.com",
  "I.Jon.1948@newhixmeusers.com",
  "G.Ree.2001@newhixmeusers.com",
  "J.Mar.2000@newhixmeusers.com",
  "G.Gre.2000@newhixmeusers.com",
  "B.Van.1955@newhixmeusers.com",
  "C.Wal.2001@newhixmeusers.com",
  "J.Ski.1974@newhixmeusers.com",
  "R.McC.1974@newhixmeusers.com",
  "S.McK.2000@newhixmeusers.com",
  "M.Dun.1970@newhixmeusers.com",
]
`;
